components/input-validation-mgt/org.wso2.carbon.identity.input.validation.mgt/src/main/java/org/wso2/carbon/identity/input/validation/mgt/model/handlers/UsernameValidationConfigurationHandler.java

```diff
@@ -0,0 +1,111 @@
+
+    @Override
+    public ValidationConfiguration getDefaultValidationConfiguration(String tenantDomain) throws
+            InputValidationMgtException {
+
+        ValidationConfiguration configuration = new ValidationConfiguration();
+        configuration.setField(USERNAME);
+        List<RulesConfiguration> rules = new ArrayList<>();
+
+        try {
+            RealmConfiguration realmConfiguration = getRealmConfiguration(tenantDomain);
+            String javaRegex = realmConfiguration.getUserStoreProperty(UserCoreConstants
+                    .RealmConfig.PROPERTY_USER_NAME_JAVA_REG_EX);
+            String jsRegex = realmConfiguration.
+                    getUserStoreProperty(UserCoreConstants.RealmConfig.PROPERTY_USER_NAME_JS_REG_EX);
+
+            // Return the JsRegex if the default regex has been updated by the user.
+            if (!javaRegex.isEmpty() && !jsRegex.isEmpty() && !DEFAULT_EMAIL_REGEX_PATTERN.equals(javaRegex)) {
+                rules.add(getRuleConfig("JsRegExValidator", JS_REGEX, jsRegex));
+                configuration.setRegEx(rules);
+            } else {
+                rules.add(getRuleConfig(EmailFormatValidator.class.getSimpleName(),
+                        ENABLE_VALIDATOR, Boolean.TRUE.toString()));
+                configuration.setRules(rules);
+            }
+            return configuration;
+        } catch (InputValidationMgtException e) {
+            throw new InputValidationMgtException(ERROR_GETTING_EXISTING_CONFIGURATIONS.getCode(), e.getMessage());
+        }
+    }
+
+    @Override
+    public boolean validateValidationConfiguration(List<RulesConfiguration> configurationList)
+            throws InputValidationMgtClientException {
+
+        List<String> validatorNames = new ArrayList<>();
+        configurationList.forEach(config -> validatorNames.add(config.getValidatorName()));
+        int validConfigurations = 0;
+        for (RulesConfiguration configuration: configurationList) {
+            if (EmailFormatValidator.class.getSimpleName().equals(configuration.getValidatorName())) {
+                if (Boolean.parseBoolean(configuration.getProperties().get(ENABLE_VALIDATOR))) {
+                    validConfigurations += 1;
+                }
+                validatorNames.remove(EmailFormatValidator.class.getSimpleName());
+            } else if (AlphanumericValidator.class.getSimpleName().equals(configuration.getValidatorName())) {
+                if (Boolean.parseBoolean(configuration.getProperties().get(ENABLE_VALIDATOR))) {
+                    if (validatorNames.contains(LengthValidator.class.getSimpleName())) {
+                        validConfigurations += 1;
+                        validatorNames.remove(LengthValidator.class.getSimpleName());
+                    }
+                }
+                validatorNames.remove(AlphanumericValidator.class.getSimpleName());
+            }
+        }
+        if (validConfigurations != 1 || !validatorNames.isEmpty()) {
```

Comment:
Why we need both validConfigurations and validatorNames here?

Reply:
To be allowed for validator combination, there must be exactly one valid combination and no other validators which not belongs to the valid combination. Eg: EmailFormatValidator-true, AlphanumericValidator-false, LengthValidator must not be allowed but EmailFormatValidator-false, AlphanumericValidator-true, LengthValidator must be allowed.